File: frontend/src/app/_login-help/login-informacije.ts
export class LoginInformacije {
  vrijednost: string=null;
  isAdministrator: boolean = false;
  isKorisnik: boolean = false;
  isRadnik: boolean = false;
  isLogiran: boolean = false;
  korisnickoIme: string = null;
  korisnikId: number;
}

export interface AutentifikacijaToken{
  id:number;
  vrijednost:string;
  korisnickiNalogId:number;
  korisnickiNalog: KorisnickiNalog;
}

export interface KorisnickiNalog {
  id: number;
  korisnickoIme: string;
  isAdministrator: boolean;
  isKorisnik: boolean;
  isRadnik: boolean;
}

File: api/e-Biblioteka/ModulAutentifikacija/Models/LoginInformacije.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace e_Biblioteka.ModulAutentifikacija.Models
{
    public class LoginInformacije
    {

        public LoginInformacije(AutentifikacijaToken autentifikacijaToken)
        {
            this.autentifikacijaToken = autentifikacijaToken;
        }

        AutentifikacijaToken autentifikacijaToken { get; set; }
        public KorisnickiNalog korisnickiNalog => autentifikacijaToken?.korisnickiNalog;

        public bool isLogiran => korisnickiNalog != null;
        public bool isPermisijaKnjigeCRUD => isLogiran && (korisnickiNalog.isRadnik || korisnickiNalog.isAdministrator);
        public bool isPermisijaRadnici => isLogiran && korisnickiNalog.isAdministrator;
        public bool isPermisijaKorisnik => isLogiran && korisnickiNalog.isKorisnik;
    }
}
